custom_mutators/aflpp/aflpp.c

```diff
@@ -1,89 +1,122 @@
 #include "afl-mutations.h"
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/msg.h>
+#include <time.h>
 
 typedef struct my_mutator {
-
   afl_state_t *afl;
   u8          *buf;
   u32          buf_size;
-
 } my_mutator_t;
 
 my_mutator_t *afl_custom_init(afl_state_t *afl, unsigned int seed) {
-
   (void)seed;
 
   my_mutator_t *data = calloc(1, sizeof(my_mutator_t));
   if (!data) {
-
     perror("afl_custom_init alloc");
     return NULL;
-
   }
-
+  // allocate MAX size buf
   if ((data->buf = malloc(MAX_FILE)) == NULL) {
-
     perror("afl_custom_init alloc");
     return NULL;
-
   } else {
-
     data->buf_size = MAX_FILE;
-
+  }
+  
+  afl->from_llm = false;
+
+  /* the mutation, send request to LLM, then receive mutate seed */
+  My_message my_msg;
+  int        msg = 200;
+
+  // Create or open the message queue
+  int msqid = msgget((key_t)1234, IPC_CREAT | 0666);
+  if (msqid == -1) {
+    perror("msgget() failed");
+    exit(1);
   }
 
-  data->afl = afl;
+  // send the request (empty message)
+  memcpy(my_msg.data_buff, &msg, sizeof(int));
+  my_msg.data_type = TYPE_REQUEST;
 
+  int snd_status = msgsnd(msqid, &my_msg, 0, 0);
+  if (snd_status == -1) {
+    perror("msgsnd() failed");
+    exit(1);
```

Comment:
I'm not sure if `exit(1)` is the best option. How do other AFL++'s code abort?

The reason is that AFL++ also holds a fork server, which, if not cleared properly, leaves an orphan process in the OS. We had this bug before, so please double-check.